src/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs

```diff
@@ -629,8 +629,31 @@ private bool StartWithCreateProcess(ProcessStartInfo startInfo)
             return ret;
         }
 
-        private static Encoding GetEncoding(int codePage)
+        private static Encoding GetConsoleEncodingOrThrow(ConsoleStreamDirection encodingType)
         {
+            const int INVALID_HANDLE = 6;
+            int codePage;
+            try
+            {
+                codePage = (encodingType == ConsoleStreamDirection.Input) ?
+                    (int)Interop.Kernel32.GetConsoleCP() :
+                    (int)Interop.Kernel32.GetConsoleOutputCP();
+
+                int error = Marshal.GetLastWin32Error();
+                if (error == INVALID_HANDLE)
```

Comment:
I am not seeing the docs saying GetConsoleCP/GetConsoleOutputCP is setting the last error 

[https://msdn.microsoft.com/en-us/library/windows/desktop/ms683162(v=vs.85).aspx](https://msdn.microsoft.com/en-us/library/windows/desktop/ms683162(v=vs.85).aspx)

how did you confirm this is the case?